src/scripts/main.js

```diff
@@ -1,3 +1,285 @@
+      return elem !== newMatrix[i][index];
+    });
+
+    if (answer) {
+      return true;
+    }
+  }
+
+  return false;
+}
+
+document.addEventListener('keydown', (e) => {
+  const arrayOfKeys = [
+    'ArrowRight',
+    'ArrowLeft',
+    'ArrowDown',
+    'ArrowUp',
+  ];
+
+  if (!arrayOfKeys.includes(e.key) || !starting) {
+    return null;
+  }
+
+  const link = deepClone(table2);
+  const rotating = transpose(link);
+
+  function arrowMove(matrix, reverse = false, vertical = false) {
+    table2 = vertical
+      ? transpose(moveCells(matrix, reverse, true))
+      : moveCells(matrix, reverse, true);
+
+    if (!onChanges(link, table2)) {
+      return null;
+    }
+    addInRandomCell(table2);
+    addValuesToCells(table2);
+    changeStyleCells(cells);
+  }
+
+  switch (e.key) {
+    case 'ArrowRight':
+      arrowMove(link);
+      break;
+    case 'ArrowLeft':
+      arrowMove(link, true);
+      break;
+    case 'ArrowDown':
+      arrowMove(rotating, false, true);
+      break;
+    case 'ArrowUp':
+      arrowMove(rotating, true, true);
+      break;
+  }
+
+  checkOnWin(table2);
+
+  for (const row of table2) {
+    if (!row.includes(0)) {
+      const checkTable = deepClone(table2);
+      const down = transpose(moveCells(rotating, true));
+      const up = transpose(moveCells(rotating, false));
+      const right = moveCells(table2, false);
+      const left = moveCells(table2, true);
+      const results = [
+        onChanges(checkTable, down),
+        onChanges(checkTable, up),
+        onChanges(checkTable, right),
+        onChanges(checkTable, left),
+      ];
+
+      if (results.every(el => el === false)) {
+        messageLose.classList.remove('hidden');
+        starting = false;
+      }
+    }
+  }
+});
+
+function transpose(original) {
+  const copy = [];
+
+  for (let i = 0; i < original.length; ++i) {
+    for (let j = 0; j < original[i].length; ++j) {
+      if (original[i][j] === undefined) {
+        continue;
+      }
+
+      if (copy[j] === undefined) {
+        copy[j] = [];
+      }
+
+      copy[j][i] = original[i][j];
+    }
+  }
+
+  return copy;
+}
+
+function moveCells(matrix, reverse = false, original = false) {
+  const copy = !(reverse)
+    ? deepClone(matrix).map(elem => elem.reverse())
+    : deepClone(matrix);
+
+  function removeZeroes(row) {
+    return row.filter(num => num !== 0);
+  }
+
+  function slide(row) {
+    let newRow = removeZeroes(row);
+
+    for (let i = 0; i < newRow.length - 1; i++) {
+      if (newRow[i] === newRow[i + 1]) {
+        newRow[i] *= 2;
+        newRow[i + 1] = 0;
+
+        if (original) {
+          addScore(newRow[i]);
+        }
+      }
+      newRow = removeZeroes(newRow);
+    }
+
+    while (newRow.length < columnLength) {
+      newRow.push(0);
+    }
+
+    return newRow;
+  }
+
+  const copy1 = [];
+
+  for (let j = 0; j < columnLength; j++) {
+    let row = copy[j];
+
+    row = slide(row);
+    copy1[j] = row;
```

Comment:
Try not to use numbers in the variable names. Better to use the words `firstCopy`, `secondCopy` for example